docs/src/modules/ROOT/pages/using-timefold-solver/running-the-solver.adoc

```diff
@@ -180,76 +209,78 @@ Any variable that changed between the "before move" solution and the "after move
 Any variable that changed between the "after move" solution and "after undo move" solution without either a
 `beforeVariableChanged` or `afterVariableChanged` would be reported here.
 
-This mode is <<environmentModeReproducible,reproducible>> (see the reproducible mode).
+This mode is <<environmentModeReproducibility,reproducible>> (see the reproducible mode).
 It is also intrusive because it calls the method `calculateScore()` more frequently than a non-assert mode.
 
 The `TRACKED_FULL_ASSERT` mode is by far the slowest mode,
 because it clones solutions before and after each move.
 
 
+[#environmentModeFullAssert]
+==== `FULL_ASSERT`
+
+The `FULL_ASSERT` mode turns on all assertions (such as assert that the incremental score calculation is uncorrupted for each move) to fail-fast on a bug in a Move implementation, a constraint, the engine itself, ...
+
+This mode is <<environmentModeReproducibility,reproducible>>.
+It is also intrusive because it calls the method `calculateScore()` more frequently than a non-assert mode.
+
+The `FULL_ASSERT` mode is horribly slow,
+because it does not rely on incremental score calculation.
+
+
 [#environmentModeNonIntrusiveFullAssert]
-=== `NON_INTRUSIVE_FULL_ASSERT`
+==== `NON_INTRUSIVE_FULL_ASSERT`
 
 The `NON_INTRUSIVE_FULL_ASSERT` turns on several assertions to fail-fast on a bug in a Move implementation,
```

Comment:
Do we have a list of the assertions somewhere? Because "ALL" (FULL_ASSERT) and "several" (NONE_INTRUSIVE_FULL_ASSERT) isn't very telling. 

Side note: having to say we only activate "several assertions" for NONE_INTRUSIVE_FULL_ASSERT is a bit weird. (But probably legacy?)

Reply:
We do not, unfortunately. Which assertions are enabled also depends on what each phase supports - listing this with all the possible combinations would IMO bring even more complexity. I'll look into rewording this a bit.